viper/parser.py

```diff
@@ -280,31 +303,48 @@ def mk_full_signature(code):
 
 # Main python parse tree => LLL method
 def parse_tree_to_lll(code, origcode):
-    _events, _defs, _globals = get_defs_and_globals(code)
+    _contracts, _events, _defs, _globals = get_contracts_and_defs_and_globals(code)
     _names = [_def.name for _def in _defs] + [_event.target.id for _event in _events]
     # Checks for duplicate funciton / event names
     if len(set(_names)) < len(_names):
         raise VariableDeclarationException("Duplicate function or event name: %s" % [name for name in _names if _names.count(name) > 1][0])
+    contracts = {}
+    # Create the main statement
+    o = ['seq']
     # Initialization function
     initfunc = [_def for _def in _defs if is_initializer(_def)]
     # Regular functions
     otherfuncs = [_def for _def in _defs if not is_initializer(_def)]
-    # Create the main statement
     sigs = {}
-    o = ['seq']
     if _events:
         for event in _events:
             sigs[event.target.id] = EventSignature.from_declaration(event)
+    for _contractname in _contracts.keys():
```

Comment:
I don't think there's a need to get a copy of the keys here, right?

This should suffice:

```python
for _contractname in _contracts:
```